smol/capp/generate/groundstate/__init__.py

```diff
@@ -0,0 +1,8 @@
+"""Ground state groundstate functionalities."""
+from importlib.util import find_spec
+
+# Check cvxpy installation.
+if find_spec("cvxpy") is None:
```

Comment:
Question, Since you import the main class in `generate/__init__.py will raising this error  not raise an error when importing other functions from `capp.generate` even when they don't depend on `cvxpy`?

Reply:
I see. Then probably we should remove ground state solver classes from smol.capp.generate's __init__.py. If the users wish to use ground state solver they'll have to import it from smol.capp.generate.groundstate.

Reply:
I think we should still have the convenience import, and have a graceful way to fail and mention to users the cvxpy is required.

Reply:
The new import way has been implemented, change the notebook as well!